apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/r4/providers/pac/AbstractR4ResourceProvider.java

```diff
@@ -282,6 +289,39 @@ private T transformEntity(
     }
   }
 
+  /**
+   * Returns the {@link Mbi} associated with the given {@link RdaFissClaim} or {@link RdaMcsClaim},
+   * depending on resource type.
+   *
+   * @param <T> generic type extending {@link IBaseResource}
+   * @param claimIdType the {@link ResourceTypeV2} indicating what the claim's type is (either fiss
+   *     or mcs)
+   * @param claimEntity the claim entity itself; either a {@link RdaFissClaim} or {@link
+   *     RdaMcsClaim}
+   * @return the {@link Mbi} associated with the given claim entity, or <code>null</code> if not
+   *     found
+   */
+  private <T extends IBaseResource> @Nullable Mbi getClaimEntityMbi(
+      ResourceTypeV2<T, ?> claimIdType, Object claimEntity) {
+    return switch (claimIdType.getTypeLabel()) {
+      case "fiss" -> ((RdaFissClaim) claimEntity).getMbiRecord();
+      case "mcs" -> ((RdaMcsClaim) claimEntity).getMbiRecord();
+      default -> null;
```

Comment:
I would probably throw an exception here since this case should never get hit.

Reply:
Agreed. I've gone ahead and done this in [64a0dbd](https://github.com/CMSgov/beneficiary-fhir-data/pull/2457/commits/64a0dbd164a5e7226697d34dd271b975eadb8351).